src/api/main.py

```
@@ -26,6 +26,98 @@
+        from src.services.config import load_config_with_main
+
+        project_root = Path(__file__).parent.parent.parent
+        main_config = load_config_with_main("main.yaml", project_root)
+        agent_config = load_config_with_main("agents.yaml", project_root)
+
+        main_tools = set(main_config.get("solve", {}).get("valid_tools", []))
+        agent_tools = set(agent_config.get("investigate", {}).get("valid_tools", []))
+
+        if not agent_tools.issubset(main_tools):
+            drift = agent_tools - main_tools
+            raise RuntimeError(CONFIG_DRIFT_ERROR_TEMPLATE.format(drift=drift))
```

Ruff (0.14.11)

Abstract `raise` to an inner function

(TRY301)